src/main/java/com/google/api/codegen/config/GapicProductConfig.java

```diff
@@ -228,6 +247,13 @@ public static GapicProductConfig create(
       throw new RuntimeException(e);
     }
 
+    String configSchemaVersion = configProto.getConfigSchemaVersion();
```

Comment:
This validation logic should not be duplicated. If the two `create` methods share logic, they should probably be refactored.

Reply:
Agree. I don't have enough knowledge to refactor these two create's into one. In theory, the validation logic can be done in GAPIC config advisor. I left two TODOs to move the checking into GAPIC config, so that there won't be two identical validation logic in this class.